bundle/config/mutator/resolve_variable_references.go

```diff
@@ -205,3 +222,43 @@ func (m *resolveVariableReferences) resolveOnce(b *bundle.Bundle, prefixes []dyn
 
 	return hasUpdates, diags
 }
+
+// If includeResources is true, the 'resources' config is mutation, otherwise, it is excluded.
```

Comment:
Rather than hidden flag, can we just explicitly list what we want to mutate?

Without resources, this is all top level keys of root dyn value.

With resource, it's just root["resources"]

Reply:
+1 that would make this simpler to read. And it forces listing the top level keys whereas it is implicit now, so seems like a win-win.

Reply:
I've added `merge.Select` and `merge.AntiSelect` and re-implemented. I think this pattern is very useful, especially if we can make it work on arbitrary paths instead of keys. The behavior remains exactly the same as today, I've just added all existing keys.